status/discover.go

```diff
@@ -0,0 +1,86 @@
+package status
+
+import (
+	"io/ioutil"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+)
+
+// The Kiali ServiceAccount token.
+var saToken string
+
+var clientFactory kubernetes.ClientFactory
+
+func DiscoverJaeger() (url string, err error) {
+	conf := config.Get()
+
+	if conf.ExternalServices.Jaeger.URL != "" {
+		return conf.ExternalServices.Jaeger.URL, nil
+	}
+	tracing := conf.ExternalServices.Jaeger.Service
+	jaeger := conf.ExternalServices.Jaeger.JaegerService
+	ns := conf.ExternalServices.Jaeger.Namespace
+	// Check if there is a Tracing service in the namespace
+	url, err = discoverUrlService(ns, tracing)
+	if err != nil || url == "" {
```

Comment:
I don't understand this part. It seems like the URL we will used can be picked up of any of "jaeger" service or "tracing" service? What use case is it intended to solve? And why wouldn't we have just 1 service name config and the user is expected to provide the one that works?

Also, I find the config names a bit confusing: "Jaeger.Service" and "Jaeger.JaegerService" it's not easy to understand what they relate to.

Reply:
This fetch if there is a tracing service first, if not we get the jaeger-query service

Reply:
My question was more about why?

Reply:
I mean, can really jaeger-query service and tracing service be used interchangeably? That seems odd.

Reply:
My concern is that there may be environments without the tracing service, so in that case we should look for jaeger service @jotak

Reply:
But if someone has an environment where Jaeger is accessible from another service name... then he would just have to put that service name in config, no? I'm still missing something.